accepted/2024/Dotnet-list-package-vulnerable-uses-auditsources.md

```diff
@@ -0,0 +1,117 @@
+Addressing this limitation is a broader issue we would like to resolve in the future (see [#13026)](https://github.com/NuGet/Home/issues/13026). 
+This proposal, however, focuses solely on adding support for `<AuditSources>`, which will utilize the VulnerabilityInfo resource of the specified sources.
+
+## Explanation
+
+### Functional explanation
+
+#### Scenario 1: User specifies both `<AuditSources>` and `<PackageSources>`
+
+**User Action:**
+
+- Specifies `<AuditSources>` and `<PackageSources>` in NuGet.config.
+- Runs `dotnet list package --vulnerable`.
+
+**Expected Behavior:**
+
+- If `<AuditSources>` contains an endpoint supporting the VulnerabilityInfo resource, NuGet will prioritize it as the primary source for vulnerability data.
+  - If the source does not support the VulnerabilityInfo resource, the command will log a warning indicating the source does not have a `VulnerabilityInfo` resource. This will follow the status quo for restore and log [NU1905](https://learn.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1905).
+- The command will iterate through all packages in the project, identify vulnerabilities, and list them along with the sources used for vulnerability information.
+- Sources(Audit sources) used in this process are logged the way `PackageSources` that are used for vulnerabilities are logged.
+- The output, including JSON format, will reflect the sources used for vulnerability data.
+
+![Image showing output of dotnet list package --vulnerable](./meta/dotnet-list-package/dlpAudiSources.png)
+
+#### Scenario 2: User specifies `<PackageSources>` only
+
+**User Action:**
+
+- Specifies `<PackageSources>` in NuGet.config.
+- Runs `dotnet list package --vulnerable`.
+
+**Expected Behavior:**
+
+- The command behaves as it currently does, using the Registration API of `<PackageSources>` to fetch vulnerability data.
+- Performs per-package HTTP requests to retrieve vulnerability information. As mentioned earlier this is an issue we would like to address in a separate [work item](https://github.com/NuGet/Home/issues/13026)
+
+### Technical explanation
+
+#### Logic Flow
+
+1. **Trigger:** The `dotnet list package --vulnerable` command is executed.
+2. **Check for `<AuditSources>`:**
+   - If `<AuditSources>` are specified, verify if they support the VulnerabilityInfo resource.
+   - Use `<AuditSources>` to load vulnerability data if supported.
+3. **Fallback:** If `<AuditSources>` are unavailable, fallback to default behavior of using PackageSources.
+4. **Warn:** If the `<AuditSources>` do not support VulnerabilityInfo resource:
+   - Log a waning [NU1905](https://learn.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1905).
+
+   > warning NU1905: Audit source 'Contoso' did not provide any vulnerability data.
+
+5. **Output:** List identified vulnerabilities and the sources used.
+
+#### Changes to our Code
```

Comment:
even in a technical spec, this is too much detail imo. 

The PR could contain this if you as the author think it'll help people review faster.